### PR TITLE
docs: bump mocha/arabica go version

### PR DESCRIPTION
**Pull request by jcstein**
## Summary by CodeRabbit

- **New Features**
	- Updated version numbers for `golangNodeMocha` and `golangNodeArabica` to reflect the latest enhancements.
  
- **Bug Fixes**
	- Ensured that versioning is consistent across components, improving overall stability.

### CODE REVIEW

**Comment by jcstein** [comment]
@rootulp does this conflict with the version required for app v2.1.2?

File: .vitepress/constants/constants.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const constants = Object.freeze({
   golangNodeMainnet: "1.22.0",
-  golangNodeMocha: "1.22.0",
-  golangNodeArabica: "1.22.0",
+  golangNodeMocha: "1.23.0",
+  golangNodeArabica: "1.23.0",
   golangApp: "1.22.0",
   golangCore: "1.22.0",
```
Comment (rootulp) [inline .vitepress/constants/constants.js:6]: @jcstein these use Go 1.22.0 which work with celestia-app v2.1.2 b/c https://github.com/celestiaorg/celestia-app/blob/48173df3dc78f9348eedb3796f29ef9e9e5dc584/go.mod#L3

We could also do this

```suggestion
  golangApp: "1.22.6",
  golangCore: "1.22.6",
```

Comment (jcstein) [reply .vitepress/constants/constants.js:6]: Build failed with 1.22 on node v0.16.0. Tagging @walldiss for review from node team to recommend either 1.23 or 1.22.6

Comment (rootulp) [reply .vitepress/constants/constants.js:6]: Can these diverge? My preference based on go.mod would be:

```
  golangNodeMocha: "1.23.0",
  golangNodeArabica: "1.23.0",
  golangApp: "1.22.6",
  golangCore: "1.22.6",
```

- https://github.com/celestiaorg/celestia-node/blob/6744f648649ebb5fee1b27faf7aca96ecf4519b2/go.mod#L3
- https://github.com/celestiaorg/celestia-app/blob/48173df3dc78f9348eedb3796f29ef9e9e5dc584/go.mod#L3

Comment (jcstein) [reply .vitepress/constants/constants.js:6]: Yep, this should work! Good suggestion

File: .vitepress/constants/constants.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const constants = Object.freeze({
   golangNodeMainnet: "1.22.0",
-  golangNodeMocha: "1.22.0",
-  golangNodeArabica: "1.22.0",
+  golangNodeMocha: "1.23.0",
+  golangNodeArabica: "1.23.0",
   golangApp: "1.22.0",
```
Comment (jcstein) [inline .vitepress/constants/constants.js:5]:
```suggestion
  golangApp: "1.22.6",
```

File: .vitepress/constants/constants.js (hunk 1 above)
Comment (jcstein) [inline .vitepress/constants/constants.js:6]:
```suggestion
  golangCore: "1.22.6",
```

**Pull request merged**